.github/workflows/scripts.yml

```diff
@@ -0,0 +1,27 @@
+name: Script unit tests
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "installer.sh"
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    name: Run script unit tests
+    runs-on: ubuntu-latest
+    # This runs on Alpine to make sure we're testing with actual sh.
```

Comment:
I'm confused, doesn't the script force things to run as bash due to the shebang? Also, Ubuntu should include Bourne shell (e.g. dash) too

Reply:
The installer shebang is `#!/bin/sh` but since that's often a symlink to bash/dash/etc I've used Alpine to make sure it's really using `sh` to catch any bashisms that might sneak in there.

Reply:
Ah! Good point, I was looking at `test-scripts.sh`, which uses bash explicitly: `#!/usr/bin/env bash`

This makes sense then. I think Ubuntu should be using dash for /bin/sh by default, though I'm not sure if GitHub Actions does some customization to the environment:

```
$ docker run --rm -it ubuntu:latest
root@081f1b5d886d:/# which sh
/usr/bin/sh
root@081f1b5d886d:/# ls -al /usr/bin/sh
lrwxrwxrwx 1 root root 4 Jul 18  2019 /usr/bin/sh -> dash
```

There's also a handy `checkbashisms` script in `devscripts` (install it with apt) that you may find useful. I use it in my dotfiles build: https://github.com/jawnsy/dotfiles/blob/0a27557f15aba599056171abc5ff936e7b530d13/.github/workflows/build.yml#L49

Reply:
Oh that's dope! I'll definitely add that.